### PR TITLE
Remove the FontSizeChanged event from TermControl

### LINKED ISSUE #1104: All TermControl/TermApp/TermConnection event handlers should be TypedEventHandlers
WinRT [design guidelines](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/author-events#design-guidelines) dictate that event handlers should have two arguments, `T` and `EventArgs`. For any class `T` that produces an event, this ensures that the event _handler_ can identify which instance of `T` fired the currently-handled event.

> The signature for an event handler delegate should consist of two parameters: sender (`IInspectable`), and args (some event argument type, for example `RoutedEventArgs`).

Given that we are exposing an internal API, we can specify a more specific `T` than `IInspectable`.

C++/WinRT supports this by way of `winrt::TypedEventHandler<T, Args...>`.

We should move all the TermConnection, TermApp and TermControl event handlers to use `TypedEventHandler`s.

### bonus

We can get rid of `DECLARE_EVENT` and `DEFINE_EVENT`, and augment `DECLARE_EVENT_WITH_TYPED_HANDLER` (and `DEFINE...TYPED`)  to use `typename class_type` as the sender type, so that we don't need to repeat ourselves:

https://github.com/microsoft/terminal/blob/880272c7483a3bb6893dbec3725d0eb7375ab78a/src/cascadia/inc/cppwinrt_utils.h#L42-L47

```diff
-#define DECLARE_EVENT_WITH_TYPED_EVENT_HANDLER(name, eventHandler, sender, args) \
+#define DECLARE_EVENT_WITH_TYPED_EVENT_HANDLER(name, eventHandler, args) \
     public: \
-    winrt::event_token name(Windows::Foundation::TypedEventHandler<sender, args> const& handler); \
+    winrt::event_token name(Windows::Foundation::TypedEventHandler<typename class_type, args> const& handler); \     void name(winrt::event_token const& token) noexcept; \
     private: \
-    winrt::event<Windows::Foundation::TypedEventHandler<sender, args>> eventHandler;
+    winrt::event<Windows::Foundation::TypedEventHandler<typename class_type, args>> eventHandler;
```

which, when used, becomes:

```diff
-DECLARE_EVENT_WITH_TYPED_EVENT_HANDLER(PasteFromClipboard, _clipboardPasteHandlers, TerminalControl::TermControl, TerminalControl::PasteFromClipboardEventArgs);
+DECLARE_EVENT_WITH_TYPED_EVENT_HANDLER(PasteFromClipboard, _clipboardPasteHandlers, TerminalControl::PasteFromClipboardEventArgs);
```

### PR DESCRIPTION
**Pull request by zadjii-msft**
I originally just wanted to close #1104, but then discovered that hey, this event wasn't even used anymore. Excerpts of Teams convo:

* [Snap to character grid when resizing window by mcpiroman · Pull Request #3181 · microsoft/terminal (github.com)](https://github.com/microsoft/terminal/pull/3181/files#diff-d7ca72e0d5652fee837c06532efa614191bd5c41b18aa4d3ee6711f40138f04c) added it to Tab.cpp
  * where it was added 
  * which called `pane->Relayout` which I don't even REMEMBER
* By [Add functionality to open the Settings UI tab through openSettings by leonMSFT · Pull Request #7802 · microsoft/terminal (github.com)](https://github.com/microsoft/terminal/pull/7802/files#diff-83d260047bed34d3d9d5a12ac62008b65bd6dc5f3b9642905a007c3efce27efd), there was seemingly no FontSizeChanged in Tab.cpp (when it got moved to terminaltab.cpp)
 

> `Pane::Relayout` functionally did nothing because sizing was switched
 to `star` sizing at some point in the past, so it was just deleted.

From [Misc pane refactoring by Rosefield · Pull Request #11373 · microsoft/terminal](https://github.com/microsoft/terminal/pull/11373/files#r736900998)

So, great. We can kill part of it, and convert the rest to a `TypedEvent`, and get rid of `DECLARE_` / `DEFINE_`.

`ScrollPositionChangedEventArgs` was ALSO apparently already promoted to a typed event, so kill that too.

### CODE REVIEW

File: src/cascadia/TerminalControl/ControlCore.cpp [hunk 1]
```diff
@@ -947,7 +946,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
 
         const auto actualNewSize = _actualFont.GetSize();
-        _FontSizeChangedHandlers(actualNewSize.width, actualNewSize.height, initialUpdate);
+        _FontSizeChangedHandlers(*this, *winrt::make_self<FontSizeChangedArgs>(actualNewSize.width, actualNewSize.height));
```
Comment (DHowett) [inline src/cascadia/TerminalControl/ControlCore.cpp:949]: `*make_self<T>()` is equivalent to `make<T>` when T's default interface is the same as the one you're converting to.

For new code, we should prefer the clearer one :)

**Review by DHowett — APPROVED** [review]

**Review by carlos-zamora — APPROVED** [review]

File: src/cascadia/TerminalControl/ControlCore.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalControl/ControlCore.cpp]: I forgot that make works even on a non projected ctor

**Review by DHowett — APPROVED** [review]
I forgot that make works even on a non projected ctor

**Comment by DHowett** [comment]
I pushed a merge so that it would kick the CI

**Review by DHowett — APPROVED** [review]

**Pull request merged**